src/data/int/gcd.lean

```diff
@@ -302,6 +302,25 @@ begin
   exact int.nat_abs_dvd_iff_dvd.mpr h
 end
 
+/-- Euclid's lemma: if `a ∣ b * c` and `gcd a c = 1` then `a ∣ b`.
+Compare with `is_coprime.dvd_of_dvd_mul_left` and
+`unique_factorization_monoid.dvd_of_dvd_mul_left_of_no_prime_factors` -/
+lemma dvd_of_dvd_mul_left_of_gcd_one {a b c : ℤ} (habc : a ∣ b * c) (hab : (gcd a c : ℤ) = 1) :
```

Comment:
Why not just use `gcd a c = 1` for the hypothesis?

Reply:
Good point, I got caught up on the fact that `int.gcd_eq_gcd_ab` uses `(gcd x y : ℤ)` and didn't look for a way around it.